src/cast_core/src/actors/RunCommand.cpp

```diff
@@ -28,11 +26,76 @@
 
 #include <boost/log/trivial.hpp>
 
+#include <cast_core/actors/RunCommand.hpp>
+
 #include <gennylib/ExecutionStrategy.hpp>
 #include <gennylib/MongoException.hpp>
 #include <gennylib/v1/RateLimiter.hpp>
 #include <gennylib/value_generators.hpp>
 
+namespace {
+
+/**
+ * @private
+ * @param command passed to run_command()
+ * @return if the command requires waiting for a stepdown/election to complete before continuing.s
+ */
+bool requiresWaitForStepdown(bsoncxx::document::view& command) {
+    return command.find("replSetStepDown") != command.end();
+}
+
+/**
+ * @private
+ * @param exception exception received from mongocxx::database::run_command()
+ * @return if the exception is a result of a connection error or timeout
+ */
+bool isTimeout(mongocxx::operation_exception& exception) {
+    auto what = std::string{exception.what()};
+    return what.find("socket error or timeout") != std::string::npos;
```

Comment:
Well, due to the ambiguity of the c-driver, this could be other things besides a timeout.

Reply:
ya...are you suggesting to change anything here though?

Reply:
Well, ideally I'd like `isTimeout` to be renamed `isNetworkFailure`.